pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -231,4 +235,27 @@ private static String getNativeConsoleEncoding() {
         }
         return null;
     }
+
+    public GlobalAnalysisListener newListener(Project project, List<String> inputPaths) throws IOException {
```

Comment:
I suppose, Formatter itself can be moved to an internal package? I will add an `@InternalApi` on the formatter class and this method

Reply:
I think the PMDTask, Formatter, SourceLanguage, and RuleSetWrapper need to stay public. This is the API towards Ant. I guess, that Ant uses reflection to call the different methods in PMDTask, e.g. when it encounters `<formatter>` in the build.xml, it will call `addFormatter` on the task - and probably creates a new Formatter.
So, it's `@InternalApi` for developers that use PMD, but is public API for ant - hope that makes sense :)

Reply:
Moving it into the internal package should work as long as this package is visible to Ant...